funsite/templates/lms/register.html

```diff
@@ -23,9 +23,8 @@
                      error: onError});
          });
 
-         function onSuccess(json) {
-             var url = json.redirect_url || "${reverse('root')}";
-             location.href = url;
+         function onSuccess() {
+            location.href = location.origin + location.search.replace('?next=', '/');
```

Comment:
Less simple but more robust WDYT ?

```suggestion
            const next = location.search.match('[?&]next=([^&]*)')?.[1] ?? '/';
            location.href = next;
```

Reply:
I used `URLSearchParams` instead of a regex.